### PR TITLE
INTYGFV-15209: date range component

### PR DESCRIPTION
**Pull request by SopraErik**
This also handles [INTYGFV-15208](https://inera.atlassian.net/browse/INTYGFV-15208) - read only view

### CODE REVIEW

File: packages/common/src/components/UnifiedView/UvDateRangeList/UvDateRangeList.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  value: ValueDateRangeList
+  config: ConfigUeSickLeavePeriod
+}> = ({ value, config }) => {
+  if (value.list.length > 0 && value.list.some((val) => val.from && val.to)) {
+    return (
+      <Table>
+        <TableHeader>
+          <TableRow>
+            <TableCell>Nedsättningsgrad</TableCell>
+            <TableCell>Från och med</TableCell>
```
Comment (UlrikaSopra) [inline packages/common/src/components/UnifiedView/UvDateRangeList/UvDateRangeList.tsx:15]: Should we use label from BE? 
image

Comment (SopraErik) [reply packages/common/src/components/UnifiedView/UvDateRangeList/UvDateRangeList.tsx:15]: DateRangeList is an old valuetype used for the sickLeave component, and in this case we don't get any label from config from backend

File: packages/common/src/components/UnifiedView/UvDateRange/UvDateRange.tsx [hunk 2]
```diff
@@ -1,37 +1,25 @@
...
-    )
-  }
-  return null
+const DateRangeWrapper = styled.div`
+  display: inline-block;
+`
+
+export const UvDateRange: React.FC<{ value: ValueDateRange; config: ConfigUeDateRange }> = ({ value, config }) => {
+  return (
+    <>
```
Comment (SamuelTilly) [inline packages/common/src/components/UnifiedView/UvDateRange/UvDateRange.tsx:12]: Fragment can be removed.

File: packages/common/src/utils/faker/fakeCertificateData.ts [hunk 3]
```diff
@@ -468,6 +470,33 @@ export const fakeDateElement = (
...
+  fakeDataElement(
+    {
+      ...data,
+      config: {
+        type: ConfigTypes.UE_DATE_RANGE,
+        fromLabel: 'Fr.o.m',
+        toLabel: 't.o.m',
+        ...data?.config,
+      },
+      // value: { type: CertificateDataValueType.DATE, date: '2022-09-29', ...data?.value },
```
Comment (SamuelTilly) [inline packages/common/src/utils/faker/fakeCertificateData.ts:486]: comment could be removed.

File: packages/webcert/src/feature/certificate/Question/QuestionUvResolve.test.tsx [hunk 4]
```diff
@@ -412,6 +421,25 @@ export function createQuestionWithMultipleDates(): CertificateDataElement {
...
+    id: '',
+    text: '',
+    type: ConfigTypes.UE_DATE_RANGE,
+    fromLabel: 'Fr.o.m',
+    toLabel: 't.o.m',
+  }
+
+  return createQuestion(value, config)
+}
+
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Question/QuestionUvResolve.test.tsx:442]: This should use the new `fakeDateRangeElement` function instead

File: packages/webcert/src/feature/certificate/UnifiedEdit/UeDateRange/Styles.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+`
+export const DateGrid = styled.div`
+  display: grid;
+  align-items: baseline;
+  grid-template-columns: 1fr 1fr;
+  @media (max-width: 1200px) {
+    grid-template-columns: 1fr;
+    grid-gap: 8px;
+  }
+`
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/UnifiedEdit/UeDateRange/Styles.tsx:27]: Move these components into `UeDateRange`  where they are used and remove `Styles.tsx`

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
There are some minor changes, but overall it looks good. However, I wonder if we could queue this PR after the changes to clientValidation since that would significantly simplify this component, and we could avoid having too many merge conflicts. https://github.com/sklintyg/frontend/pull/644

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
Is it possible to only highlight the date that is missing an value?

image

**Comment by SopraErik** [comment]
> Is it possible to only highlight the date that is missing an value when validating?
> 
> image

I will take a renewed look at this once 15177 is merged, since that includes som major updates to how det datepicker works

**Comment by SopraErik** [comment]
> > Is it possible to only highlight the date that is missing an value when validating?
> > image
> 
> I will take a renewed look at this once 15177 is merged, since that includes som major updates to how det datepicker works

This is fixed now

**Review by SamuelTilly — APPROVED** [review]
Very nice!

**Review by ChrilleSigerhed — APPROVED** [review]
Well done!

**Pull request merged**